tutorials/tensordictmodule.ipynb

```diff
@@ -0,0 +1,1020 @@
+   "id": "3be0fafd",
+   "metadata": {},
+   "source": [
+    "# TensorDictModule"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "94bd315a",
+   "metadata": {},
+   "source": [
+    "We recommand reading the TensorDict tutorial before going through this one."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0652352c",
+   "metadata": {},
+   "source": [
+    "For a convenient usage of the `TensorDict` class with `nn.Module`, TorchRL provides an interface between the two named `TensorDictModule`. <br/>\n",
+    "The `TensorDictModule` class is an `nn.Module` that takes a `TensorDict` as input when called. <br/>\n",
+    "It is up to the user to define the keys to be read as input and output."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "129a6de9-cf97-4565-a229-c05ad18df882",
+   "metadata": {},
+   "source": [
+    "## `TensorDictModule` by examples"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "5b0241ab",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torch\n",
+    "import torch.nn as nn\n",
+    "\n",
+    "from torchrl.data import TensorDict\n",
+    "from torchrl.modules import TensorDictModule, TensorDictSequence"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9d1c188a",
+   "metadata": {},
+   "source": [
+    "### Example 1: Simple usage"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1d21a711",
+   "metadata": {},
+   "source": [
+    "Let's suppose we have `TensorDict` with 2 entries `\"a\"` and `\"b\"` but only the value associated with `\"a\"` has to be read by the network."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "id": "6f33781f",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "TensorDict(\n",
+      "    fields={\n",
+      "        a: Tensor(torch.Size([5, 3]), dtype=torch.float32),\n",
+      "        a_out: Tensor(torch.Size([5, 10]), dtype=torch.float32),\n",
+      "        b: Tensor(torch.Size([5, 4, 3]), dtype=torch.float32)},\n",
+      "    batch_size=torch.Size([5]),\n",
+      "    device=cpu,\n",
+      "    is_shared=False)\n"
+     ]
+    }
+   ],
+   "source": [
+    "tensordict = TensorDict(\n",
+    "    {\"a\": torch.randn(5, 3), \"b\": torch.zeros(5, 4, 3)},\n",
+    "    batch_size=[5],\n",
+    ")\n",
+    "linear = TensorDictModule(\n",
+    "    nn.Linear(3, 10), in_keys=[\"a\"], out_keys=[\"a_out\"]\n",
+    ")\n",
+    "linear(tensordict)\n",
+    "assert (tensordict.get(\"b\") == 0).all()\n",
+    "print(tensordict)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "00035cbd",
+   "metadata": {},
+   "source": [
+    "### Example 2: Multiple inputs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "06a20c22",
+   "metadata": {},
+   "source": [
+    "Suppose we have a slightly more complex network that takes 2 entries and averages them into a single output tensor."
```

Comment:
We could add something like "To make a `TensorDictModule` instance read multiple input values, one must register them in the `in_keys` keyword argument of the constructor."